src/TestFramework/Coverage/BufferedSourceFileFilter.php

```diff
@@ -76,10 +76,16 @@ public function __construct(
         }
     }
 
+    /**
+     * We duck-typed the input to be iterable<SplFileInfo|Trace> in the interface,
+     * but we actually only use iterable<Trace> here.
+     */
     public function filter(iterable $input): iterable
     {
         return take($this->filter->filter($input))
-            ->filter(function (Trace $trace): bool {
+            ->filter(function ($trace): bool {
+                Assert::isInstanceOf($trace, Trace::class);
```

Comment:
The idea to mingle Trace with SplFileInfo bit us in the butt here: the other option would be to make traces actually implement SplFileInfo, or go away from SplFileInfo entirely. Or something else.